script.js

```diff
@@ -0,0 +1,356 @@
+          this._fireEvent('scrollbox-scroll', {
+            element: this
+          });
+        }
+        break;
+      case 'scroll':
+        // Gecko may scroll the scrollbox automatically, we want to prevent this
+        // behavior inorder to have correct view.
+        evt.target.scrollLeft = 0;
+        break;
+    }
+  };
+
+  /**
+   * Get the node element of the item element
+   */
+  proto._getNodeFromItem = function(itemElem) {
+    if (!itemElem) {
+      return null;
+    }
+
+    var nodeElem = itemElem;
+    // Find the node element. If the node is removed from scrollbox, it's parent
+    // will be null, so we may handle the case here.
+    while (nodeElem.parentElement && nodeElem.parentElement !== this) {
+      nodeElem = nodeElem.parentElement;
+    }
+
+    return nodeElem;
+  };
+
+  /**
+   * Scroll the list to the item element
+   */
+  proto._scrollTo = function(itemElem) {
+    this.translateX = this._getScrollOffset(this._getNodeFromItem(itemElem));
+    this.listElem.style.transform = 'translateX(' + this.translateX + 'px)';
+  };
+
+
+  /**
+   * Get the offset of the node element
+   */
+  proto._getScrollOffset = function(nodeElem) {
+    var nodeLeft = nodeElem.offsetLeft;
+    var nodeWidth = nodeElem.offsetWidth;
+    var listWidth = this.listElem.offsetWidth;
+    var newTranslate = this.translateX;
+
+    if (listWidth < this.viewWidth) {
+      // scroll to center
+      return (this.viewWidth - listWidth) / 2 + this.margin;
+    } else if (nodeLeft + nodeWidth >
+                          -this.translateX + this.viewWidth + this.margin) {
+      // scroll left
+      newTranslate = this.viewWidth - nodeLeft - nodeWidth + this.margin;
+    } else if (nodeLeft < -this.translateX + this.margin) {
+      // scroll right
+      newTranslate = -nodeLeft + this.margin;
+    }
+
+    // If the new scroll offset contains first/last node, we have to align the
+    // list to begin/end.
+    if (this.lastNode.offsetLeft + this.lastNode.offsetWidth <=
+                          -newTranslate + this.viewWidth + this.margin) {
+      return this.viewWidth + this.margin -
+                          this.lastNode.offsetLeft - this.lastNode.offsetWidth;
+    } else if (this.firstNode.offsetLeft >= -newTranslate + this.margin) {
+      return -this.firstNode.offsetLeft + this.margin;
+    }
+    return newTranslate;
+  };
+
+  /**
+   * Handle focus from spacial navigator
+   */
+  proto._handleFocus = function(itemElem) {
+    var nodeElem = this._getNodeFromItem(itemElem);
+    this._scrollTo(itemElem);
+    this._fireEvent('focus', {
+      itemElem: itemElem,
+      nodeElem: nodeElem,
+      index: Array.prototype.indexOf.call(this.children, nodeElem)
+    });
+
+    // Since we may have many elements in one node, node-changed event happens
+    // when active node is changed.
+    if (this.currentNode && this.currentNode !== nodeElem) {
+      this.currentNode.classList.remove('active');
```

Comment:
Is there any reason that we need to addd `active` class to currentNode?? If no, we shouldn't do it.